neo/Consensus/ConsensusContext.cs

```diff
@@ -279,7 +346,24 @@ public void Reset(byte viewNumber)
                     NextConsensus = Blockchain.GetConsensusAddress(NativeContract.NEO.GetValidators(Snapshot).ToArray()),
                     ConsensusData = new ConsensusData()
                 };
+                var pv = Validators;
                 Validators = NativeContract.NEO.GetNextBlockValidators(Snapshot);
+                if (_witnessSize == 0 || (pv != null && pv.Length != Validators.Length))
+                {
+                    // Compute the expected size of the witness
+                    using (ScriptBuilder sb = new ScriptBuilder())
+                    {
+                        for (int x = 0; x < M; x++)
+                        {
+                            sb.EmitPush(new byte[64]);
+                        }
+                        _witnessSize = new Witness

```

Comment:
This is specially bad if we dont drop it, after cleaning next context, and eventually.number of nodes increases (but we keep same previous _witnessSize). It's better if we attain to a simple const: Witness.GetMaxSize() or Witness.MaxSize.